data/abi/abi_encode.go

```diff
@@ -204,12 +204,8 @@ func inferToSlice(value interface{}) ([]interface{}, error) {
 	if reflectVal.Kind() != reflect.Slice && reflectVal.Kind() != reflect.Array {
 		return nil, fmt.Errorf("cannot infer an interface value as a slice of interface element")
 	}
-	if reflectVal.IsNil() {
-		if reflectVal.Kind() == reflect.Slice {
-			return nil, nil
-		}
-		return nil, fmt.Errorf("cannot infer nil value for array kind interface")
-	}
+	// * if input is a slice, with nil, then reflectVal.Len() == 0
```

Comment:
This is not always true. Since value is an `interface{}`, it can be nil in two different ways:
1. Either the interface is assigned to no value (truly nil), e.g. `var value interface{} = nil`
2. Or it is assigned to a value that is nil, e.g. `var value interface{} = []uint64(nil)`

From my testing, `.Len()` panics in the first case but returns 0 in the second case.

Reply:
The first case cannot get through the previous `Kind` check, which is `invalid` rather than `Slice` or `Array`.

The second case you suggesting has kind `Slice`, and it does have length can be computable.

Reply:
Good point, I missed that in my late night review 😅 

This change looks good to me